src/HypothesisClient/Serializer/AnnotationDenormalizer.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace eLife\HypothesisClient\Serializer;
+
+use DateTimeImmutable;
+use eLife\HypothesisClient\Model\Annotation;
+use eLife\HypothesisClient\Model\Annotation\Document;
+use eLife\HypothesisClient\Model\Annotation\Permissions;
+use eLife\HypothesisClient\Model\Annotation\Target;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+final class AnnotationDenormalizer implements DenormalizerInterface, DenormalizerAwareInterface
+{
+    use DenormalizerAwareTrait;
+
+    public function denormalize($data, $class, $format = null, array $context = []) : Annotation
+    {
+        $data['document'] = $this->denormalizer->denormalize($data['document'], Document::class);
+        $data['target'] = $this->denormalizer->denormalize($data['target'][0], Target::class);
```

Comment:
should we assert that `$data['target']` only has 1 element, or should we silently ignore any additional element in it?